doc/source/whatsnew/v0.22.0.txt

```diff
@@ -151,7 +151,7 @@ Indexing
 - Bug in :class:`IntervalIndex` where empty and purely NA data was constructed inconsistently depending on the construction method (:issue:`18421`)
 - Bug in ``IntervalIndex.symmetric_difference()`` where the symmetric difference with a non-``IntervalIndex`` did not raise (:issue:`18475`)
 - Bug in indexing a datetimelike ``Index`` that raised ``ValueError`` instead of ``IndexError`` (:issue:`18386`).
-
+- Bug in ``TimeDeltaBlock._can_hold_element()`` where masked assignment of a timedelta series converts the series values to float64 (:issue:`18493`)
```

Comment:
you can move to 0.21.1

don't use internal language here, these are user-level notes. So say something like

Bug in masked assignment of a ``timedelta64[ns]`` dtype ``Series`` with a ``Timedelta`` incorrectly coerced to float.